src/platform/packages/private/kbn-data-streams/src/integration_tests/client.test.ts

```diff
@@ -293,26 +298,30 @@ describe('DataStreamClient', () => {
       await DataStreamClient.initialize({
         logger,
         elasticsearchClient,
-        dataStreams: testDataStream,
+        dataStream: testDataStream,
       });
 
+      const sameVersionMappings = {
+        properties: { somethingElse: mappings.text() },
+      } satisfies MappingsDefinition;
+
       await DataStreamClient.initialize({
         logger,
         elasticsearchClient,
-        dataStreams: {
+        dataStream: {
           ...testDataStream,
           version: 1, // same version as initial deployment
           template: {
             ...testDataStream.template,
-            mappings: { properties: { somethingElse: mappings.text() } as any }, // some new mappings
+            mappings: sameVersionMappings, // some new mappings
           },
         },
       });
 
       expect(getIndexTemplateSpy).toHaveBeenCalledTimes(2);
-      expect(putIndexTemplateSpy).toHaveBeenCalledTimes(1);
+      expect(putIndexTemplateSpy).toHaveBeenCalledTimes(1); // Index template not updated when version is same
       expect(createDataStreamSpy).toHaveBeenCalledTimes(1);
-      expect(putMappingSpy).toHaveBeenCalledTimes(0); // No mapping updates were made
+      expect(putMappingSpy).toHaveBeenCalledTimes(1); // Mappings are still applied to write index even when version is same
```

Comment:
This was intentional design decision to force developers to create version N+1 of their definition otherwise we will not update mappings. The idea is that you will keep old definitions for testing, but maybe you have something else in mind?